Add documentation for duplicate webapp route and error responses

mollykreis opened the pull request:
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/systemlink-OpenAPI-documents/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

- Add documentation for the new duplicate webapp route that will be added to the webapp service.
- Add documentation for the error response format. (This documentation was adapted directly from the Alarm service)
- Update Dynamic LINQ support documentation

### Why should this Pull Request be merged?

Documentation is good.

### What testing has been done?

Verified the document renders correctly in the swagger editor.

mollykreis committed:
Add documentation for duplicate webapp route and add error responses

scotia673 commented on webapp/niapp.yaml:
```diff
@@ -235,7 +255,32 @@ paths:
+        - $ref: '#/parameters/DuplicateWebAppRequest'
+      responses:
+        200:
+          description: Success
```
A successful request should include the newly-created webapp's metadata in the response right? I would expect the documentation to outline a response in the success case similar to the way the successful response is described for the create route.

mollykreis committed:
more updates

- Add success output for duplicate
- Add ability to specify workspace during duplicate
- Remove documentation for Contains() in LINQ queries

scotia673 reviewed (dismissed):
Thanks for updating the success field, Molly!

spanglerco commented on webapp/niapp.yaml:
```diff
@@ -235,7 +255,32 @@ paths:
+        default:
+          $ref: '#/responses/Error'
 
+  '/webapps/duplicate/{sourceId}':
```
At first I was thinking this should be a `POST /duplicate-webapps` to match our HTTP conventions, but then I realized sourceId is a path parameter and that there's already a `POST /webapps/{id}/duplicate/{sourceId}`. Would it makes sense to merge those two routes into a `POST /webapps/{id}/duplicate` where source ID is in the path and destination ID is optional in the body?

That way, the URL is implying the "duplicate" action on the `/webapps/{id}` resource.

mollykreis replied:
We could do that. I'm not sure what it would take to drop support for the existing route, though. We would probably need to keep it for at least a little while. Do you have an issue keeping both routes for now?

spanglerco replied:
No, that sounds fine to me.

mollykreis replied:
After an offline conversation with @spanglerco, we decided there wasn't a pressing reason to try to have the new duplicate route be dual-purpose to handle specifying an ID to duplicate only the content or specifying a name + workspace to duplicate content + metadata. This simplifies the documentation, testing, and implementation.

I will, however, change this new route to be `/webapps/{sourceId}/duplicate`.

mollykreis committed:
webapps/duplicate/{id} to webapps/{id}/duplicate
